stellargraph/utils/random.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import random
+import tensorflow as tf
+
+
+_sg_seed = None
+
+
+def get_seed(seed):
+    """
+    Convenience function to use the global seed by default if the provided seed is None.
+
+    Args:
+        seed (int, optional): seed value
+
+    Returns:
+        seed if not None, otherwise the global seed
+
+    """
+    if seed is None:
+        return _sg_seed
+    else:
+        return seed
+
+
+def set_seed(seed, set_np_seed=True, set_tf_seed=True, set_random_seed=True):
+    """
+    Set the seed for all possible randomness in StellarGraph. Note that this
+    also sets the global random seed for the following external modules:
+        * numpy
+        * tensorflow
+        * random
+
+    Args:
+        seed (int, optional): seed value
+        set_np_seed (bool, default True): If true, mutate the global numpy seed
```

Comment:
FWIW, the function signature already lists the defaults:

![image](https://user-images.githubusercontent.com/1203825/72944576-6593da80-3dcd-11ea-8c10-b83daafdb2d3.png)

https://external-builds.readthedocs.io/html/stellargraph/705/api.html#stellargraph.utils.random.set_seed

So maybe

```suggestion
        set_np_seed (bool, optional): If true, mutate the global numpy seed
```

and for the others; or maybe not necessary?

Reply:
I think we've discussed this before somewhere else, but I thought `something, optional` just meant a parameter is of type `Optional[something]`, rather than a parameter with a default value which is written as `x, default something`? I guess this might just be purely whatever convention we prefer, but my point at the time was how would you indicate that something is of optional type in the docstring if you reserved `optional` for things having default values?

Reply:
e.g. I intended `get_seed` to be an optional type, but not have a default value of `None`